### PR TITLE
Introduce KAS Anchor option 

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes

This PR added KAS anchor option.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -592,6 +592,39 @@ var (
 		Name:  "anchoring",
 		Usage: "Enable anchoring for service chain",
 	}
+	// KAS
+	ServiceChainKASAnchorFlag = cli.BoolFlag{
```
Comment (ehnuje) [inline cmd/utils/flags.go:596]: As the namespace begins with `kas`, how about changing the prefix of variables from `ServiceChainKAS` to `KASServiceChain`?

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -592,6 +592,39 @@ var (
...
+		Name:  "kas.anchor",
```
Comment (ehnuje) [inline cmd/utils/flags.go:597]: How about adding `sc` between `kas` and the flag name?
```suggestion
		Name:  "kas.sc.anchor",
```

File: cmd/utils/flags.go (hunk 1 above)
Comment (KimKyungup) [reply cmd/utils/flags.go]: Reflected

File: cmd/utils/flags.go (hunk 2 above)
Comment (KimKyungup) [reply cmd/utils/flags.go]: Reflected

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -592,6 +592,39 @@ var (
...
+	}
+	KASServiceChainAnchorPeriodFlag = cli.Uint64Flag{
+		Name:  "kas.sc.anchor.period",
+		Usage: "The period to anchor service chain blocks to KAS",
+		Value: 1,
+	}
+	KASServiceChainAnchorUrlFlag = cli.StringFlag{
+		Name:  "kas.sc.anchor.url",
+		Usage: "The url for KAS anchor",
+		Value: "http://anchor.wallet-api.dev.klaytn.com/v1/anchor", // TODO-Klaytn need to change production server
```
Comment (KimKyungup) [inline cmd/utils/flags.go:608]:
```suggestion
```

File: cmd/utils/flags.go [hunk 4]
```diff
@@ -592,6 +592,39 @@ var (
...
+	}
+	KASServiceChainAnchorOperatorFlag = cli.StringFlag{
+		Name:  "kas.sc.anchor.operator",
+		Usage: "The operator address for KAS anchor",
+	}
+	KASServiceChainXKRNFlag = cli.StringFlag{
+		Name:  "kas.xkrn",
+		Usage: "The x-krn for KAS",
+		Value: "krn:1001:anchor:test:operator-pool:op1",
```
Comment (KimKyungup) [inline cmd/utils/flags.go:617]:
```suggestion
```

**Comment by KimKyungup** [comment]
@aidan-kwon @ehnuje I removed KAS server information. PTAL. :)

File: cmd/kscn/main.go [hunk 5]
```diff
@@ -76,6 +76,14 @@ var scnHelpFlagGroups = []utils.FlagGroup{
 			utils.VTRecoveryIntervalFlag,
 			utils.ServiceChainNewAccountFlag,
 			utils.ServiceChainAnchoringFlag,
+			// KAS
+			utils.KASServiceChainAnchorFlag,
+			utils.KASServiceChainAnchorPeriodFlag,
+			utils.KASServiceChainAnchorUrlFlag,
+			utils.KASServiceChainAnchorOperatorFlag,
+			utils.KASServiceChainAccessKeyFlag,
+			utils.KASServiceChainSecreteKeyFlag,
```
Comment (jeongkyun-oh) [inline cmd/kscn/main.go:85]: `utils. KASServiceChainSecreteKeyFlag` -> `utils. KASServiceChainSecretKeyFlag`

File: cmd/ksen/main.go [hunk 6]
```diff
@@ -74,6 +74,14 @@ var senHelpFlagGroups = []utils.FlagGroup{
 			utils.VTRecoveryFlag,
 			utils.VTRecoveryIntervalFlag,
 			utils.ServiceChainAnchoringFlag,
+			// KAS
+			utils.KASServiceChainAnchorFlag,
+			utils.KASServiceChainAnchorPeriodFlag,
+			utils.KASServiceChainAnchorUrlFlag,
+			utils.KASServiceChainAnchorOperatorFlag,
+			utils.KASServiceChainAccessKeyFlag,
+			utils.KASServiceChainSecreteKeyFlag,
```
Comment (jeongkyun-oh) [inline cmd/ksen/main.go]:
```suggestion
			utils.KASServiceChainSecretKeyFlag,
```

File: cmd/kspn/main.go [hunk 7]
```diff
@@ -77,6 +77,14 @@ var spnHelpFlagGroups = []utils.FlagGroup{
 			utils.VTRecoveryFlag,
 			utils.VTRecoveryIntervalFlag,
 			utils.ServiceChainAnchoringFlag,
+			// KAS
+			utils.KASServiceChainAnchorFlag,
+			utils.KASServiceChainAnchorPeriodFlag,
+			utils.KASServiceChainAnchorUrlFlag,
+			utils.KASServiceChainAnchorOperatorFlag,
+			utils.KASServiceChainAccessKeyFlag,
+			utils.KASServiceChainSecreteKeyFlag,
```
Comment (jeongkyun-oh) [inline cmd/kspn/main.go]:
```suggestion
			utils.KASServiceChainSecretKeyFlag,
```

File: cmd/utils/flags.go [hunk 8]
```diff
@@ -592,6 +592,37 @@ var (
...
+	}
+	KASServiceChainXKRNFlag = cli.StringFlag{
+		Name:  "kas.xkrn",
+		Usage: "The x-krn for KAS",
+	}
+	KASServiceChainAccessKeyFlag = cli.StringFlag{
+		Name:  "kas.accesskey",
+		Usage: "The access key id for KAS",
+	}
+	KASServiceChainSecreteKeyFlag = cli.StringFlag{
```
Comment (jeongkyun-oh) [inline cmd/utils/flags.go]:
```suggestion
	KASServiceChainSecretKeyFlag = cli.StringFlag{
```

File: cmd/utils/nodecmd/dumpconfigcmd.go [hunk 9]
```diff
@@ -284,6 +284,35 @@ func makeServiceChainConfig(ctx *cli.Context) (config sc.SCConfig) {
...
+		if cfg.KASAnchorOperator == "" {
+			logger.Crit("KAS anchor operator should be set", "key", utils.KASServiceChainAnchorOperatorFlag.Name)
+		}
+
+		cfg.KASAccessKey = ctx.GlobalString(utils.KASServiceChainAccessKeyFlag.Name)
+		if cfg.KASAccessKey == "" {
+			logger.Crit("KAS access key should be set", "key", utils.KASServiceChainAccessKeyFlag.Name)
+		}
+
+		cfg.KASSecreteKey = ctx.GlobalString(utils.KASServiceChainSecretKeyFlag.Name)
```
Comment (jeongkyun-oh) [inline cmd/utils/nodecmd/dumpconfigcmd.go:306]:
```suggestion
		cfg.KASSecretKey = ctx.GlobalString(utils.KASServiceChainSecretKeyFlag.Name)
```

File: cmd/utils/nodecmd/dumpconfigcmd.go [hunk 10]
```diff
@@ -284,6 +284,35 @@ func makeServiceChainConfig(ctx *cli.Context) (config sc.SCConfig) {
...
+		if cfg.KASSecreteKey == "" {
```
Comment (jeongkyun-oh) [inline cmd/utils/nodecmd/dumpconfigcmd.go:307]:
```suggestion
		if cfg.KASSecretKey == "" {
```

File: node/sc/config.go [hunk 11]
```diff
@@ -94,6 +94,15 @@ type SCConfig struct {
...
 	VTRecoveryInterval uint64
 	Anchoring          bool
+
+	// KAS
+	KASAnchor         bool
+	KASAnchorUrl      string
+	KASAnchorPeriod   uint64
+	KASAnchorOperator string
+	KASAccessKey      string
+	KASSecreteKey     string
```
Comment (jeongkyun-oh) [inline node/sc/config.go:104]:
```suggestion
	KASSecretKey     string
```

File: node/sc/gen_config.go [hunk 12]
```diff
@@ -30,6 +30,13 @@ func (s SCConfig) MarshalTOML() (interface{}, error) {
 		ParentChainID         uint64
 		VTRecovery            bool
 		VTRecoveryInterval    uint64
+		Anchoring             bool
+		KASAnchor             bool
+		KASAnchorPeriod       uint64
+		KASAnchorOperator     string
+		KASAccessKey          string
+		KASSecreteKey         string
```
Comment (jeongkyun-oh) [inline node/sc/gen_config.go:38]:
```suggestion
		KASSecretKey         string
```

File: node/sc/gen_config.go [hunk 13]
```diff
@@ -53,6 +60,13 @@ func (s SCConfig) MarshalTOML() (interface{}, error) {
 	enc.ParentChainID = s.ParentChainID
 	enc.VTRecovery = s.VTRecovery
 	enc.VTRecoveryInterval = s.VTRecoveryInterval
+	enc.Anchoring = s.Anchoring
+	enc.KASAnchor = s.KASAnchor
+	enc.KASAnchorPeriod = s.KASAnchorPeriod
+	enc.KASAnchorOperator = s.KASAnchorOperator
+	enc.KASAccessKey = s.KASAccessKey
+	enc.KASSecreteKey = s.KASSecreteKey
```
Comment (jeongkyun-oh) [inline node/sc/gen_config.go:68]:
```suggestion
	enc.KASSecretKey = s.KASSecretKey
```

**Comment by KimKyungup** [comment]
Thanks. @jeongkyun-oh Done. 👍

**Comment by ehnuje** [comment]
gofmt and linter failed 😢

**Pull request merged**